cpg_infra/abstraction/azure.py

```diff
@@ -54,43 +246,88 @@ def create_bucket(
         versioning: bool = True,
         project: str = None,
     ) -> Any:
+        # Policies are set at the storage account level
+        # the rules are generated using other function calls
+        # e.g. bucket_rule_archive
+        # They are then passed here to apply to a particular 'bucket'
+        # (azure blob container)
+        # So first we modify to filter the rule to apply only to the
+        # new bucket
+        bucket_filter = az.storage.ManagementPolicyFilterArgs(
+            prefix_match=[name], blob_types=['blockBlob']
+        )
+
+        def apply_filter(rule):
+            rule.definition.filters = bucket_filter
+            rule.name = f'{name}-{rule.name}'
+            return rule
+
+        lifecycle_rules = filter(lambda x: x, lifecycle_rules)
+        lifecycle_rules = list(map(apply_filter, lifecycle_rules))
+
+        self.storage_account_lifecycle_rules.extend(lifecycle_rules)
+
         return az.storage.BlobContainer(
-            f'bucket-{name}',
-            resource_group_name=self.resource_group_name,
-            account_name=self.storage_account_name,
+            f'{name}',
+            account_name=self.storage_account.name,
+            resource_group_name=project or self.resource_group.name,
             container_name=name,
+            metadata={'bucket': name},
+            # TODO: work out requester_pays in Azure
         )
 
+    def bucket_membership_to_role(self, membership: BucketMembership):
+        if membership in (BucketMembership.MUTATE, BucketMembership.APPEND):
+            return f'{self.subscription}/providers/Microsoft.Authorization/roleDefinitions/ba92f5b4-2d11-453d-a403-e96b0029c9fe'
```

Comment:
Where does the last guid come from, could we make it a constant or add a comment?